bool cartesian
bool joint
int8 current_controlled_joint